### PR TITLE
Remove TS3.6 support from babel-types

### PR DESCRIPTION
**Pull request by sandersn**
TS3.6 is no longer supported on DT, so babel-types doesn't need to support it anymore either.

I copied the contents of the .ts files in ts3.6 to the top of their counterparts in the root, then deleted them.

### CODE REVIEW

**Comment by sandersn** [comment]
Nope, turns out I have more work to do (1) to dedupe multiple function declarations (2) to remove import types that reference `./ts3.6`

File: types/babel-types/index.d.ts [hunk 1]
```diff
@@ -7,7 +7,1997 @@
...
+export function assertTSTypeParameterDeclaration(node: object | null | undefined, opts?: object): void;
+export function assertTSTypeParameterInstantiation(node: object | null | undefined, opts?: object): void;
+export function assertTSTypePredicate(node: object | null | undefined, opts?: object): void;
+export function assertTSTypeQuery(node: object | null | undefined, opts?: object): void;
+export function assertTSTypeReference(node: object | null | undefined, opts?: object): void;
+export function assertTSUndefinedKeyword(node: object | null | undefined, opts?: object): void;
+export function assertTSUnionType(node: object | null | undefined, opts?: object): void;
+export function assertTSVoidKeyword(node: object | null | undefined, opts?: object): void;
 
 export function assertArrayExpression(node: object | null | undefined, opts?: object): asserts node is import('./ts3.6/index').ArrayExpression;
```
Comment (SimonSchick) [inline types/babel-types/index.d.ts:2002]: I think you missed those references.

**Review by SimonSchick — CHANGES_REQUESTED** [review]

**Comment by SimonSchick** [comment]
Also I already got a PR here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54895 I just need to rebase it, or do you plan to do this in multiple staged?

**Comment by sandersn** [comment]
Woops, I didn't know about that your PR included all the ts3.6 directories. I thought you were just working on node. Let's close my two PRs in favour of yours.

**Pull request closed without merging**